app/scripts/modules/core/src/pipeline/config/triggers/pipeline/PipelineTrigger.tsx

```diff
@@ -1,64 +1,60 @@
 import * as React from 'react';
 import Select, { Option } from 'react-select';
 
-import { Observable } from 'rxjs';
+import { Observable, Subject } from 'rxjs';
 
-import { ApplicationReader } from 'core/application/service/ApplicationReader';
-import { BaseTrigger } from 'core/pipeline';
+import { Application, ApplicationReader } from 'core/application';
+import { BaseTrigger, PipelineConfigService } from 'core/pipeline';
 import { IPipeline, IPipelineTrigger } from 'core/domain';
-import { PipelineConfigService } from 'core/pipeline/config/services/PipelineConfigService';
+import { Omit } from 'core/presentation';
 import { Checklist } from 'core/forms';
-import { Application } from '@spinnaker/core';
-import { Subject } from 'rxjs/Subject';
+
+type IPipelineTriggerConfig = Omit<IPipelineTrigger, 'parentExecution' | 'parentPipelineId' | 'user' | 'rebake'>;
 
 export interface IPipelineTriggerConfigProps {
   status: string[];
   trigger: IPipelineTrigger;
   application: Application;
   pipelineId: string;
-  triggerUpdated: (trigger: IPipelineTrigger) => void;
+  triggerUpdated: (trigger: IPipelineTriggerConfig) => void;
```

Comment:
this is fine for now because the thing that passes in `triggerUpdated` is a vanilla JS file 🤷‍♂